### PR TITLE
feature/index_target

### PR DESCRIPTION
**Pull request by Rodrigo-Miranda-0**
#### Board:
* [Ticket #mn8AOCqE](https://trello.com/c/mn8AOCqE/4-3-as-a-user-i-should-be-able-to-see-all-my-targets-in-the-map-so-i-can-easily-identify-them)
---
#### Description:

* Allows for the user to see all his created targets
---
#### Notes:
*
---
#### Tasks:
- [x] Index user targets
---
#### Risk:
*
---
#### Preview:

### CODE REVIEW

File: spec/requests/api/v1/targets/index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+describe 'Get targets', type: :request do
+  subject { get api_v1_targets_path, headers:, as: :json }
+
+  let(:user) { create(:user) }
+  let(:headers) { user.create_new_auth_token }
+  let!(:targets) { create_list(:target, 3, user:) }
+
+  context 'when user is logged in' do
```
Comment (enzofab91) [inline spec/requests/api/v1/targets/index_spec.rb:11]: Can you validate that only targets from the logged in user are being returned?

**Review by jpascual1994 — APPROVED** [review]

**Review by juanmferreira93 — APPROVED** [review]

File: spec/requests/api/v1/targets/index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+describe 'Get targets', type: :request do
+  subject { get api_v1_targets_path, headers:, as: :json }
+
+  let(:user) { create(:user) }
+  let(:user2) { create(:user) }
```
Comment (enzofab91) [inline spec/requests/api/v1/targets/index_spec.rb]: Maybe?

```suggestion
  let(:another_user) { create(:user) }
```

File: spec/requests/api/v1/targets/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  let(:headers) { user.create_new_auth_token }
+  let!(:targets) { create_list(:target, 3, user:) }
+  let!(:targets2) { create_list(:target, 3, user: user2) }
```
Comment (enzofab91) [inline spec/requests/api/v1/targets/index_spec.rb]: Same here:

```suggestion
  let!(:another_targets) { create_list(:target, 3, user: user2) }
```

File: spec/requests/api/v1/targets/index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  context 'when user A is logged in' do
```
Comment (enzofab91) [inline spec/requests/api/v1/targets/index_spec.rb]:
```suggestion
  context 'when user is logged in' do
```

File: spec/requests/api/v1/targets/index_spec.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    it 'returns all user A targets' do
```
Comment (enzofab91) [inline spec/requests/api/v1/targets/index_spec.rb]: You can test that when user A is logged in, targets from other users are not being returned. Something like this:

`expect(result['targets'].pluck('title')).not_to include(other_user_targets.pluck([:id]))`

File: spec/requests/api/v1/targets/index_spec.rb [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      subject
+      result = JSON.parse(response.body)
+      expect(result['targets'].pluck('title')).to match_array(targets.pluck(:title))
+    end
+  end
+
+  context 'when user B is logged in' do
```
Comment (enzofab91) [inline spec/requests/api/v1/targets/index_spec.rb]: I think that this is not necessary if you add the other test.

**Review by enzofab91 — COMMENTED** [review]
Good work! 💯

**Review by juanmferreira93 — APPROVED** [review]

**Review by enzofab91 — APPROVED** [review]
Good work 💯

**Pull request merged**